pkgs/by-name/ki/kiwitalk/package.nix

```
@@ -0,0 +1,128 @@
+{ lib
+, fetchFromGitHub
+, copyDesktopItems
+, stdenv
+, stdenvNoCC
+, rustPlatform
+, cargo
+, cargo-tauri
+, rustc
+, openssl
+, libayatana-appindicator
+, webkitgtk
+, pkg-config
+, makeDesktopItem
+, jq
+, moreutils
+, nodePackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "kiwitalk";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "KiwiTalk";
+    repo = "KiwiTalk";
+    rev = "v${version}";
+    hash = "sha256-Th8q+Zbc102fIk2v7O3OOeSriUV/ydz60QwxzmS7AY8=";
+  };
+
+  sourceRoot = src.name;
+
+  postPatch = ''
+    substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
+      --replace "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
+  '';
+
+  pnpm-deps = stdenvNoCC.mkDerivation {
+    pname = "${pname}-pnpm-deps";
+    inherit src version;
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      nodePackages.pnpm
+    ];
+
+    installPhase = ''
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      # pnpm config set strict-ssl false
+      pnpm install --frozen-lockfile --ignore-script
+
+      rm -rf $out/v3/tmp
+      for f in $(find $out -name "*.json"); do
+        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+        jq --sort-keys . $f | sponge $f
+      done
+    '';
+
+    dontBuild = true;
+    dontFixup = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-LJPjWNpVfdUu8F5BMhAzpTo/h6ax7lxY2EESHj5P390=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "tauri-plugin-log-0.0.0" = "sha256-8BrFf7vheMJIaZD0oXpi8V4hmUJFzHJmkcRtPL1/J48=";
+      "tauri-plugin-single-instance-0.0.0" = "sha256-8BrFf7vheMJIaZD0oXpi8V4hmUJFzHJmkcRtPL1/J48=";
+    };
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    cargo
+    rustc
+    cargo-tauri
+    nodePackages.pnpm
+    copyDesktopItems
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+    libayatana-appindicator
+    webkitgtk
+  ];
+
+  preBuild = ''
+    export HOME=$(mktemp -d)
+    pnpm config set store-dir ${pnpm-deps}
+    pnpm install --offline --frozen-lockfile --ignore-script
+    pnpm rebuild
+    cargo tauri build -b deb
+  '';
+
+  preInstall = ''
+    mv target/release/bundle/deb/*/data/usr/ $out
+  '';
+
+  postInstall = ''
+    install -DT backend/bin/app/icons/512x512.png $out/share/icons/hicolor/512x512/apps/kiwi-talk.png
+    install -DT backend/bin/app/icons/256x256.png $out/share/icons/hicolor/256x256/apps/kiwi-talk.png
+    install -DT backend/bin/app/icons/128x128.png $out/share/icons/hicolor/128x128/apps/kiwi-talk.png
+    install -DT backend/bin/app/icons/32x32.png $out/share/icons/hicolor/32x32/apps/kiwi-talk.png
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "KiwiTalk";
+      exec = "kiwi-talk";
+      icon = "kiwi-talk";
+      desktopName = "KiwiTalk";
+      comment = meta.description;
```

```suggestion
      comment = "An UNOFFICIAL cross-platform KakaoTalk client";
```
a change in meta should not trigger a rebuild